finality-aleph/src/network/mod.rs

```diff
@@ -126,6 +126,13 @@ pub trait RequestBlocks<B: Block>: Clone + Send + Sync + 'static {
     /// Clear all pending justification requests. We need this function in case
     /// we requested a justification for a block, which will never get it.
     fn clear_justification_requests(&self);
+
+    /// Are we in the process of downloading the chain?
+    ///
+    /// Like [`NetworkService::is_major_syncing`][1].
+    ///
+    /// [1]: sc_network::NetworkService::is_major_syncing
+    fn is_major_syncing(&self) -> bool;
```

Comment:
Question/suggestion: is this method necessary here? It doesn't look for me like it belong here, so perhaps we shouldn't extend this trait with some ad-hoc method like this. Or perhaps some other consumers of this trait are going to care about its syncing state?

Reply:
I suggested adding this here, so perhaps I'll explain my reasoning. <_<" I thought it would fit here relatively well as far as responsibilities of components go – one component is responsible for syncing blocks, and information about whether we are in a major sync is part of these responsibilities. On the other hand I agree the API users are only interested in parts of the API, but I'm also not sure it will stay that way.

Reply:
How about creating a new trait that extends `RequestBlock` (or perhaps composition fits here better) - only party.rs would see that new method. It also looks like an implementation detail of how networking is implemented currently, users of this - I would even consider having a separate trait for something `low-level` like this.

Reply:
I'm fine with either approach (as well as leaving it as-is).

I don't think it's strictly (at most partially) an implementation detail – whatever component takes care of block syncing is extremely likely to be aware of "major syncing" or a similar state regardless of implementation. The fact that it's part of network is an implementation detail though.